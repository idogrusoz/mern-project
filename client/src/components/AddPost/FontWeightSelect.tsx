import { MenuItem, Select } from "@material-ui/core";
import React, { Dispatch, FunctionComponent, SetStateAction, useEffect } from "react";

type FontWeightSelectProps = {
    fontWeightValues: number[];
    fontWeight: number;
    setFontWeight: Dispatch<SetStateAction<number>>;
};

const FontWeightSelect: FunctionComponent<FontWeightSelectProps> = ({
    fontWeight,
    fontWeightValues,
    setFontWeight,
}) => {
    useEffect(() => {
        if (fontWeightValues.indexOf(fontWeight) === -1) {
            setFontWeight(fontWeightValues[0]);
        }
    }, [fontWeight, setFontWeight, fontWeightValues]);
    return (
        <Select fullWidth onChange={(e) => setFontWeight(Number(e.target.value))} value={fontWeight}>
            {fontWeightValues.map((value: number, i: number) => {
                return (
                    <MenuItem value={value} key={i}>
                        {value}
                    </MenuItem>
                );
            })}
        </Select>
    );
};

export default FontWeightSelect;
