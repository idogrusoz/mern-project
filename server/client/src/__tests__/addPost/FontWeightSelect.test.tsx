import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import FontWeightSelect from "../../components/addPost/FontWeightSelect";

describe("FontWeightSelect test", () => {
    const props = {
        fontWeight: 400,
        setFontWeight: jest.fn(() => {}),
        fontWeightValues: [400, 600],
    };

    const wrapper = shallow(<FontWeightSelect {...props} />);
    it("renders without crash", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
