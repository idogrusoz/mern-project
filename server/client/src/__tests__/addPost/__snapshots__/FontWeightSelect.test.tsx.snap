// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontWeightSelect test renders without crash 1`] = `
<WithStyles(ForwardRef(Select))
  fullWidth={true}
  onChange={[Function]}
  value={400}
>
  <WithStyles(ForwardRef(MenuItem))
    key="0"
    value={400}
  >
    400
  </WithStyles(ForwardRef(MenuItem))>
  <WithStyles(ForwardRef(MenuItem))
    key="1"
    value={600}
  >
    600
  </WithStyles(ForwardRef(MenuItem))>
</WithStyles(ForwardRef(Select))>
`;
